### PR TITLE
Additional GP migration updates for May 2023

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**
This submission contains the following enhancements to the GP migration.

**Migrate email addresses with Vendor and Customer addresses**
Include the email address with Vendor and Customer addresses if they are valid email addresses.

**Retain Company migration configuration settings record after migration**
The Company config settings was being deleted after the Company is migrated, but it has since been learned that it's better to keep it for troubleshooting purposes. The record will remain but will be hidden from the configuration page if the Company has been migrated.

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/Migration/Customers/GPCustomerAddress.Table.al [hunk 1]
```diff
@@ -97,6 +100,14 @@ table 4048 "GP Customer Address"
             if (CopyStr(ShipToAddress."Fax No.", 1, 14) = '00000000000000') then
                 ShipToAddress."Fax No." := '';
 
+            if GPSY01200.Get('CUS', CUSTNMBR, ADRSCODE) then
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Customers/GPCustomerAddress.Table.al:103]: Can we move the CUS to a label or a local method, so it is not hardcoded inline

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al [hunk 2]
```diff
@@ -387,13 +390,25 @@ codeunit 4022 "GP Vendor Migrator"
         OrderAddress."Fax No." := HelperFunctions.CleanGPPhoneOrFaxNumber(GPVendorAddress.FAXNUMBR);
         OrderAddress."Post Code" := GPVendorAddress.ZIPCODE;
         OrderAddress.County := GPVendorAddress.STATE;
+
+        if GPSY01200.Get('VEN', Vendor."No.", AddressCode) then
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al:394]: Same comment here about hardcoding inline

File: Apps/W1/HybridGP/app/src/pages/HybridCloudWizardExtension.PageExt.al [hunk 3]
```diff
@@ -73,28 +73,9 @@ pageextension 4014 "Hybrid Cloud Wizard Extension" extends "Hybrid Cloud Setup W
...
-                    CleanupSettingsForCompaniesPreviouslyMigrated();
+                if Rec."Product ID" = 'DynamicsGP' then
                     Page.Run(Page::"GP Migration Configuration");
-                end;
             end;
-
         }
     }
-
-    local procedure CleanupSettingsForCompaniesPreviouslyMigrated()
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/pages/HybridCloudWizardExtension.PageExt.al:85]: We need to test this well. It should include enabling and disabling the cloud migration. We should also test connecting a different DB to bring in a company

**Review by nikolakukrika — DISMISSED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 11137.

**Pull request closed without merging**